src/rules/noConsecutiveBlankLinesRule.ts

```diff
@@ -30,27 +30,21 @@ class NoConsecutiveBlankLinesWalker extends Lint.SkippableTokenAwareRuleWalker {
     public visitSourceFile(node: ts.SourceFile) {
         super.visitSourceFile(node);
 
-        // starting with 1 to cover the case where the file starts with two blank lines
-        let newLinesInARowSeenSoFar = 1;
-        Lint.scanAllTokens(ts.createScanner(ts.ScriptTarget.ES5, false, ts.LanguageVariant.Standard, node.text), (scanner: ts.Scanner) => {
-            const startPos = scanner.getStartPos();
-            if (this.tokensToSkipStartEndMap[startPos] != null) {
-                // tokens to skip are places where the scanner gets confused about what the token is, without the proper context
-                // (specifically, regex, identifiers, and templates). So skip those tokens.
-                scanner.setTextPos(this.tokensToSkipStartEndMap[startPos]);
-                newLinesInARowSeenSoFar = 0;
-                return;
+        // find all the lines that are blank or only contain whitespace
+        let blankLineIndexes: number[] = [];
+        node.getFullText().split(/\n/).forEach(function(txt, i){
+            if (txt.trim() === "") {
+                blankLineIndexes.push(i);
             }
+        });
 
-            if (scanner.getToken() === ts.SyntaxKind.NewLineTrivia) {
-                newLinesInARowSeenSoFar += 1;
-                if (newLinesInARowSeenSoFar >= 3) {
-                    const failure = this.createFailure(scanner.getStartPos(), 1, Rule.FAILURE_STRING);
-                    this.addFailure(failure);
-                }
-            } else {
-                newLinesInARowSeenSoFar = 0;
+        // now only throw failures for the fisrt number from groups of consecutive blank line indexes
+        for (let i = 0; i < blankLineIndexes.length; i++) {
+            let diff = Math.abs(blankLineIndexes[i + 1] - blankLineIndexes[i]);
+            let prevDiff = Math.abs(blankLineIndexes[i] - blankLineIndexes[i - 1]);
```

Comment:
Nice! This code is a lot more friendly, I like the simplifications. I don't feel like accessing `blankLineIndexes[-1]` and `blankLineIndexes[length]` is great though, maybe we should handle those boundary cases specially?